source/agora/node/FullNode.d

```diff
@@ -322,7 +322,9 @@ public class FullNode : API
 
     protected bool acceptBlock(const ref Block block) @trusted
     {
-        return this.ledger.acceptBlock(block);
+        this.ledger.acceptBlock(block);
+        // Maybe acceptBlock returned false for block already in ledger
+        return this.ledger.getBlockHeight() >= block.header.height;
```

Comment:
```suggestion
        return this.ledger.acceptBlock(block) &&
            // Maybe acceptBlock returned false for block already in ledger
            this.ledger.getBlockHeight() >= block.header.height;
```

Wouldn't that be more correct ?

Reply:
Same result, but yes.

Reply:
On second thought, if we receive a bad block from a node for a height we already have, do we want to fetch from more nodes?

Reply:
I added better documentation this time.